### PR TITLE
Add some docs for Blazor scalability

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Ported over from https://github.com/aspnet/specs/pull/316. I'm not super sure how or if we Ignitor and our benchmarking adventures fit in this. Would we want to consider writing a separate blog post about it?

**Comment by guardrex** [comment]
Given that there are style, grammar, metadata, file name, cross-links (in other docs), and TOC updates for this, it's quickest if I copy the content into a new PR. Do you want me to do that before or after technical review by @rynowak and @danroth27 review?

Via the new *Circuits* section, is this meant to resolve [Document "circuits" for Blazor server-side (aspnet/AspNetCore.Docs \#13360)](https://github.com/aspnet/AspNetCore.Docs/issues/13360)?

Is everything Preview 8 friendly so that we can merge as soon as done? If not, I'll direct the PR to the staging branch.

**Comment by pranavkm** [comment]
> it's quickest if I copy the content into a new PR. 

@guardrex that sounds great. There's some additional content in https://github.com/aspnet/specs/pull/316 if that helps.

**Comment by guardrex** [comment]
Sure thing. I'll wait until they review this and sign off on it.

**Comment by Rick-Anderson** [comment]
@danroth27  @rynowak  ping

File: aspnetcore/blazor/hosting-models.md [hunk 1]
```diff
@@ -77,65 +77,99 @@ There are downsides to server-side hosting:
 
 &dagger;The *blazor.server.js* script is served from an embedded resource in the ASP.NET Core shared framework.
 
+## Server-side Blazor fundamentals
+
+### Comparison with server-rendered UI
+
+One way to understand Blazor Server app is by understanding how it differs from traditional models for rendering UI in ASP.NET Core applications using Razor views or Razor pages. Both models use the Razor language to describe HTML contents, but signficantly differ in how it's rendered.
```
Comment (danroth27) [inline aspnetcore/blazor/hosting-models.md]:
```suggestion
One way to understand Blazor Server app is by understanding how it differs from traditional models for rendering UI in ASP.NET Core applications using Razor views or Razor pages. Both models use the Razor language to describe HTML content, but significantly differ in how they are rendered.
```

File: aspnetcore/blazor/hosting-models.md [hunk 2]
```diff
@@ -77,65 +77,99 @@ There are downsides to server-side hosting:
...
+
+### UI Latency
+
+UI latency is the time it takes for an action to be initiated, to the time the UI is updated. Smaller values for UI latency are imperative for an application to feel responsive. In a Blazor Server app application, each action is sent to the server, processed, and a UI diff is sent back. Consequently UI latency is the sum of network latency as well as the server latency in processing the action.
+
+For a line of business application that is limited to your private corporate network, the effects of network latency are likely imperceptible. For an application deployed over the public internet, this may become noticeable particularly if your users are geographically distributed.
+
+From our profiling, we think most Blazor server apps are going to be memory bound and memory is going to be other contributor to application latency. More memory use results in more frequent garbage collection or paging memory to disk, both of which degrade application performance and consequently increase UI latency.
+
+In our measurements, a value of 250ms or less provides a good UI experience. See <xref:best-practices> for ways to measure network latency.
```
Comment (danroth27) [inline aspnetcore/blazor/hosting-models.md]:
```suggestion
Blazor Server apps should be optimized to minimize UI latency by reducing network latency and memory usage. See <xref:best-practices> for ways to measure network latency.
```

File: aspnetcore/blazor/hosting-models.md [hunk 3]
```diff
@@ -77,65 +77,99 @@ There are downsides to server-side hosting:
...
+
+## Configuring Blazor Server application
```
Comment (danroth27) [inline aspnetcore/blazor/hosting-models.md]: Remove

File: aspnetcore/blazor/server-side-blazor-best-practices.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
+---
+title: Server Blazor app best practices
```
Comment (danroth27) [inline aspnetcore/blazor/server-side-blazor-best-practices.md]: This doc seems like a mix of stuff. Could we instead put the guidance in the relevant doc? For example, the "Harden JS interop" section would go in the JS interop doc, "Guard against multiple dispatches" would go in the components doc, etc.? @guardrex thoughts?

Comment (guardrex) [reply aspnetcore/blazor/server-side-blazor-best-practices.md]: Sure thing. I'll try to place the content into relevant topics on the new PR with the usual docs updates. I think I can get to this tomorrow after Pre9 mop-up work.

**Comment by guardrex** [comment]
Moved to https://github.com/aspnet/AspNetCore.Docs/pull/14189.

File: aspnetcore/blazor/server-side-blazor-best-practices.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        }
+    }
+}
+```
+
+### Avoid chatty events
+
+DOM events such as `oninput` or `onscroll` can produce a lot of data. Avoid using these events in your Blazor server app.
+
+### Measure network latency
```
Comment (danroth27) [inline aspnetcore/blazor/server-side-blazor-best-practices.md]: I think it's hard to argue that network latency is a security issue. How about we add this to the Blazor Server host & deploy doc instead?

Comment (guardrex) [reply aspnetcore/blazor/server-side-blazor-best-practices.md]: Ok ... I should get the cross-ref to where it was generally discussed ... perf is security. Let me see if I can find that.

**Pull request closed without merging**